benchmarking/evaluations/scenarios/cosc341/concentrate_timeslot_diversify_all_min_2_diversify_student_level.py

```diff
@@ -0,0 +1,160 @@
+from typing import List
+
+from api.models.enums import (
+    Gender,
+    DiversifyType,
+    ScenarioAttribute,
+    TokenizationConstraintDirection,
+)
+from api.models.tokenization_constraint import TokenizationConstraint
+from benchmarking.evaluations.goals import DiversityGoal, WeightGoal
+from benchmarking.evaluations.interfaces import Scenario, Goal
+
+
+class ConcentrateTimeSlotDiversifyAllGenderMin2DiversifyYearLevel(Scenario):
+    # This scenario diversifies all genders with min 2 to better compare with group matcher, which guarantees no individual will have solo status
+    def __init__(
+        self,
+        max_num_choices: int,
+        value_of_male: int = Gender.MALE,
+        value_of_female: int = Gender.FEMALE,
+        value_of_non_binary: int = Gender.NON_BINARY,
+        value_of_non_answer: int = Gender.NA,
+    ):
+        super().__init__()
+        self.value_of_female = value_of_female
+        self.max_num_choices = max_num_choices
```

Comment:
Only applies to timeslot goal so name is a little misleading

If there is only 1 run for this then you can just hard code the number of choices here without passing it in (applies to all)